submissions/VBystrov/friends-app/js/index.js

```diff
@@ -0,0 +1,326 @@
+const endpoint = 'https://randomuser.me/api/';
+const includeFields = `inc=${[
+  'gender',
+  'name',
+  'location',
+  'picture',
+  'dob',
+  'email',
+].join(',')}`;
+const amountFriends = `results=${50}`;
+const options = `?${[includeFields, amountFriends].join('&')}`;
+const requestFriendsUrl = `${endpoint}${options}`;
+const initialSearchParameters = {
+  name: null,
+  age: null,
+  gender: 'all',
+  sorting: 'nameAZ',
+};
+
+class FriendsApp {
+  constructor() {
+    this.init(requestFriendsUrl);
+  }
+
+  init(url) {
+    fetch(url)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        return response.json();
+      })
+      .then((friendsData) => {
+        const friendlist = new Friendlist(friendsData.results);
+        const search = new Search(friendlist);
+      })
+      .catch((error) => {
+        console.error(
+          'There has been a problem with your fetch operation:',
+          error
+        );
+      });
+  }
+}
+
+class Friendlist {
+  constructor(friendsData) {
+    this.container = document.getElementById('friendlist');
+    this.createCards(friendsData);
+  }
+
+  createCards(friendlistData) {
+    this.cards = [];
+    friendlistData.forEach((cardData) => {
+      this.cards.push(new Card(cardData));
+    });
+  }
+
+  applySearch(search) {
+    this.applyFilters(search);
+    this.applySorting(search.sorting);
+  }
+
+  applyFilters(search) {
+    this.filteredCards = [...this.cards];
+
+    if (search.age) {
+      this.filteredCards = this.filteredCards.filter(
+        (card) => card.age === search.age
+      );
+    }
+
+    if (search.name) {
+      this.filteredCards = this.filteredCards.filter((card) =>
+        `${card.firstName} ${card.lastName}`.includes(search.name)
+      );
+    }
+
+    if (search.gender !== 'all') {
+      this.filteredCards = this.filteredCards.filter(
+        (card) => card.gender === search.gender
+      );
+    }
+  }
+
+  applySorting(sortingOrder) {
+    let sortFunction;
+    switch (sortingOrder) {
+      case 'nameAZ':
+      case 'nameZA':
+        sortFunction = (function () {
+          const orderSign = sortingOrder === 'nameAZ' ? 1 : -1;
+          return (card1, card2) => {
+            const fullName1 = card1.firstName + card1.lastName;
+            const fullName2 = card2.firstName + card2.lastName;
+            let order = orderSign;
+            if (fullName1 < fullName2) {
+              return order * -1;
+            }
+            if (fullName1 > fullName2) {
+              return order * 1;
+            }
+            if (fullName1 === fullName2) {
+              return 0;
+            }
```

Comment:
Move this function into some variables.

And then map them.

Also, no need in part with `order`. You can do it simpler by changing the order of passed arguments.

Reply:
I can`t change order of passed arguments because i pass sortFunction to Array.sort(), so I need to change passed function. 

> Move this function into some variables.

> And then map them.

I don`t quite understand, what do you mean, but I tried to improve this part. 
I changed my mind, maybe its not worth to have one function that checks which is order every time when comparing array elements, but better to have two simpler but similar functions. What do you think?

Reply:
Left comment. You did a good job. But I think we have some place for improvements :)